lib/spack/spack/cmd/env.py

```diff
@@ -446,6 +448,40 @@ def env_deactivate(args):
     sys.stdout.write(cmds)
 
 
+#
+# env track
+#
+def env_track_setup_parser(subparser):
+    """track an environment from a directory in Spack"""
+    subparser.add_argument("-n", "--name", help="custom environment name")
+    subparser.add_argument("dir", help="path to environment")
+    arguments.add_common_arguments(subparser, ["yes_to_all"])
+
+
+def env_track(args):
+    src_path = os.path.abspath(args.dir)
+    if not ev.is_env_dir(src_path):
+        msg = f"cannot track environment {src_path} doesn't contain an environment"
+        raise ev.SpackEnvironmentError(msg)
+
+    if args.name:
+        name = args.name
+    else:
+        name = os.path.basename(src_path)
+
+    dst_path = ev.environment_dir_from_name(name, exists_ok=False)
+
+    try:
+        symlink(src_path, dst_path)
+    except SymlinkError as exc:
+        msg = f"cannot track the environment {src_path} unable to create symlink"
+        raise ev.SpackEnvironmentError(msg) from exc
```

Comment:
I don't think you need to catch and wrap this. It's likely rare and you can just allow the error to be raised directly.  What *will* happen is something the user can likely understand: a tracked/managed environment with that name already exists. 

I would check whether an existing environment is there first.  So after you check the name, see if the `dst_path` exists, and if it's a link say it's already being tracked (and where), and if it is a directory say that a managed environment by that name already exists, then fail gracefully here.